### PR TITLE
Fixed cross platform build issue.

### PR DESCRIPTION
**Pull request by namanguptagit**
## 📝 Description
Five examples (foxglove, orb_detector, ros-z-nodes, smol_vlm_video, smol_vlm2_video) unconditionally import V4L (Video4Linux) types, causing the workspace to fail compilation on macOS and Windows.
This PR gates the V4L usage in each example behind #[cfg(target_os = "linux")] at both the Cargo.toml dependency level and the source level. On non-Linux platforms, affected binaries print a clear error message and exit. For smol_vlm_video and smol_vlm2_video, only the webcam capture path is gated - the --video-file path remains fully cross-platform.
No changes to core crates, CI configuration, or pixi tasks were needed.
**⚠️ Issue Link Required**: This PR must be linked to an approved and assigned issue. See [Contributing Guide](CONTRIBUTING.md#pull-request) for details.
https://github.com/kornia/kornia-rs/issues/813
**Fixes/Relates to:** #813

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made

- Gate V4L-dependent examples behind #[cfg(target_os = "linux")] to fix compilation on macOS/Windows
-  Move V4L feature deps to [target.'cfg(target_os = "linux")'.dependencies] in example Cargo.toml files
-  Add helpful non-Linux stub main() functions with clear error messages
-  For smol_vlm_video/smol_vlm2_video, preserve the --video-file path cross-platform while gating only the webcam (V4L) path
-  Gate camera_node module in ros-z-nodes/src/lib.rs behind cfg(target_os = "linux")
- Add macOS cross-platform CI check to [rust_lint.yml](cci:7://file:///Users/namangupta/Documents/GitHub/kornia-rs/.github/workflows/rust_lint.yml:0:0-0:0) using pixi (full workspace check)
- Add Windows cross-platform CI check to [rust_lint.yml](cci:7://file:///Users/namangupta/Documents/GitHub/kornia-rs/.github/workflows/rust_lint.yml:0:0-0:0) using plain `cargo check` (core crates)

Affected examples:

- examples/foxglove — full main gated
- examples/orb_detector — full main gated
- examples/ros-z-nodes — camera_node module + ros-z-camera binary gated
- examples/smol_vlm_video — video_demo module + webcam branch gated, video_file_demo remains cross-platform
- examples/smol_vlm2_video — video_demo module + webcam branch gated, video_file_demo remains cross-platform

---

## 🧪 How Was This Tested?
- [ ] **Unit Tests:** (List new/updated tests)
- [ ] **Manual Verification:** 
1. cargo fmt --all -- --check — passes (zero formatting issues)
2. cargo clippy --workspace --no-deps --all-targets --features ci -- -D warnings  passes (zero warnings)
3. cargo check --workspace --all-targets --features ci  - passes on macOS (zero warnings)
4. cargo test --features ci  - all tests pass

- [ ] **Performance/Edge Cases:** No runtime performance impact  changes are purely compile-time cfg gates
---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] This PR strictly implements what the linked issue describes (no scope creep)
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

## 💭 Additional Context
The kornia-io crate already correctly gates V4L behind `#[cfg(all(feature = "v4l", target_os = "linux"))]`, but the example crates were importing V4L types unconditionally. Since CI runs exclusively on ubuntu-latest, this was never caught. These changes ensure the entire workspace compiles cleanly on macOS and Windows.
No changes were needed to kornia/Cargo.toml (ci feature) or pixi.toml - Cargo correctly handles platform-gated optional dependencies, so --features ci (which includes v4l) works on all platforms without issues.

CI improvements :
- macOS : Uses pixi + `pixi run rust-check` → full `cargo check --workspace --all-targets --features ci`. Catches any unconditional Linux-only imports across the entire workspace.
- Windows : Uses `cargo check` (core crates via `default-members`). pixi doesn't support Windows in this project, so this validates core library crates only.

### CODE REVIEW

**Comment by sidd-27** [comment]
how would we improve ci to catch these things beforehand?

**Comment by namanguptagit** [comment]
This was undetected because CI never compiles on those platforms. A simple cargo check on macOS/Windows would have caught this immediately.
Proposed Solution : Add a new job to rust_lint.yml with a build matrix .
If its okay i would create a pr for that seperately .
@sidd-27

**Comment by sidd-27** [comment]
sure

**Comment by namanguptagit** [comment]
Okay then what about this pr should we merge it ?

**Comment by sidd-27** [comment]
actually make it here only, fix overall comparability probelm in one pr

**Comment by namanguptagit** [comment]
Alright let me do it .

**Comment by namanguptagit** [comment]
@sidd-27 i Have updated the code please check and updated pr description for the change as well

**Comment by namanguptagit** [comment]
@sidd-27 a gentle follow up , can you check this please .

**Comment by namanguptagit** [comment]
Hi @sidd-27 can you check if its ok we can release this one

File: examples/smol_vlm2_video/src/main.rs [hunk 1]
```diff
@@ -46,6 +47,18 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+            video_demo::video_demo(&args)
+        }
+        #[cfg(not(target_os = "linux"))]
+        {
+            eprintln!(
+                "error: webcam capture (camera_id={}, fps={}) requires Video4Linux and is only supported on Linux.",
+                args.camera_id, args.fps
+            );
+            eprintln!("hint: use --video-file to process a video file instead.");
+            std::process::exit(1);
```
Comment (sidd-27) [inline examples/smol_vlm2_video/src/main.rs]: proper error handling here

**Comment by namanguptagit** [comment]
@sidd-27 fixed the issue

File: examples/ros-z-nodes/src/bin/ros-z-camera.rs [hunk 2]
```diff
@@ -23,6 +28,7 @@ struct Args {
     fps: u32,
 }
 
+#[cfg(target_os = "linux")]
 #[tokio::main]
```
Comment (sidd-27) [inline examples/ros-z-nodes/src/bin/ros-z-camera.rs]: excessive cfg usage, can we not just make a linux impl block and cfg that?

**Comment by namanguptagit** [comment]
I have addressed the change in required files @sidd-27 .

**Comment by namanguptagit** [comment]
@sidd-27 can you take a look . Thanks

**Comment by namanguptagit** [comment]
@sidd-27 Is there anything else required in this pr ?

**Comment by namanguptagit** [comment]
a gentle follow up @sidd-27

**Review by sidd-27 — APPROVED** [review]
